src/DynamicProjectOptions.cmake

```diff
@@ -78,6 +78,8 @@ macro(dynamic_project_options)
       "ENABLE_CPPCHECK\;OFF\;${MAKEFILE_OR_NINJA}\;Enable cppcheck analysis during compilation"
       "ENABLE_INTERPROCEDURAL_OPTIMIZATION\;OFF\;OFF\;Enable whole-program optimization (e.g. LTO)"
       "ENABLE_NATIVE_OPTIMIZATION\;OFF\;OFF\;Enable the optimizations specific to the build machine (e.g. SSE4_1, AVX2, etc.)."
+      "DISABLE_EXCEPTIONS\;OFF\;OFF\;Disable Exceptions (no-exceptions flag)"
+      "DISABLE_RTTI\;OFF\;OFF\;Disable RTTI (no-rtti flag)"
```

Comment:
These are interesting options for embedded targets. There's a third one that comes to mind `-fno-threadsafe-statics`.

Reply:
The reason I mentioned that flag is because it may not be desirable (or possible at all) to have mutexes (or some other form of critical sections / disabling interrupts) when static variables are accessed on some systems. A quick google tells that MSVC has a `/Zc:threadSafeInit-` flag.

I took an interests in your pull request because I'm trying to do similar stuff. @aminya is probably in a better position to judge.

Reply:
Since this optimization is unsafe, I prefer that we do not add it. In general, It would be important to not hinder the safety/correctness of a program for minor performance improvements.

Reply:
If you are sure that one part of the code doesn't need to be thread-safe, you should use the thread-local attributes to specify that specific variable. You cannot assume that this is true for all the programs and your dependencies.
https://en.wikipedia.org/wiki/Thread-local_storage#C_and_C++